File: eGlamHeelHangout/eGlamHeelHangout.Service/Database/Order.cs
using System;
using System.Collections.Generic;

namespace eGlamHeelHangout.Service.Database
{
    public partial class Order
    {
        public Order()
        {
            OrderItems = new HashSet<OrderItem>();
        }

        public int OrderId { get; set; }
        public int UserId { get; set; }
        public decimal TotalPrice { get; set; }
        public string OrderStatus { get; set; } = null!;
        public DateTime? OrderDate { get; set; }
        public string PaymentMethod { get; set; } = null!;

        public virtual User User { get; set; } = null!;
        public virtual ICollection<OrderItem> OrderItems { get; set; }
        public string? FullName { get; set; }
        public string? Email { get; set; }
        public string? Address { get; set; }
        public string? City { get; set; }
        public string? PostalCode { get; set; }
        public string? PhoneNumber { get; set; }

    }
}

File: eGlamHeelHangout/UI/eglamheelhangout_mobile/lib/models/order.dart
import 'package:json_annotation/json_annotation.dart';
import 'orderitem.dart';

part 'order.g.dart';
@JsonSerializable()
class Order {
  final int? orderId;
  final double totalPrice;
  final String orderStatus;
  final String paymentMethod;
  final String? username;
  final DateTime? orderDate;
  final List<OrderItem> items;
  final String? fullName;
  final String? email;
  final String? address;
  final String? city;
  final String? postalCode;
  final String? phoneNumber;

  Order({
    required this.orderId,
    required this.totalPrice,
    required this.orderStatus,
    required this.paymentMethod,
    required this.items,
    this.fullName,
    this.email,
    this.address,
    this.city,
    this.postalCode,
    this.username,
    this.orderDate,
    this.phoneNumber,
  });

  factory Order.fromJson(Map<String, dynamic> json) => _$OrderFromJson(json);
  Map<String, dynamic> toJson() => _$OrderToJson(this);
}
